test/mocks/server/listener_component_factory.h

```diff
@@ -4,16 +4,20 @@
 #include "envoy/server/drain_manager.h"
 #include "envoy/server/listener_manager.h"
 
+#include "source/server/listener_manager_impl.h"
+
 #include "test/mocks/network/mocks.h"
+#include "test/mocks/server/instance.h"
 
 #include "gmock/gmock.h"
 
 namespace Envoy {
 namespace Server {
-class MockListenerComponentFactory : public ListenerComponentFactory {
+
+class MockProdListenerComponentFactory : public ProdListenerComponentFactory {
```

Comment:
Why do we need to change this?
It's better to make mocked objects of the interfaces/abstract types rather than the Prod classes (one would like to use a mock of the general listener component factory)

Reply:
Some tests, like : https://github.com/envoyproxy/envoy/blob/b58eb9519b6b3cf0facd0250898fdbc2ec2353ca/test/server/listener_manager_impl_test.h#L350
to use it to initialize listener_manager_:
https://github.com/envoyproxy/envoy/blob/b58eb9519b6b3cf0facd0250898fdbc2ec2353ca/test/server/listener_manager_impl_test.h#L78

And the below code is eventually called in such test: 

https://github.com/envoyproxy/envoy/blob/b58eb9519b6b3cf0facd0250898fdbc2ec2353ca/source/server/listener_impl.cc#L617

And is trying to look for the tcp_filter_config_provider_manager, newly added in ProdListenerComponentFactory.  Without this change, such code path crash since no tcp_listener_config_provider_manager exist if instance listener_manager_ with ListenerComponentFactory.

Reply:
Right, we should mock ListenerComponentFactory.  Backed out this change.